documentation/how-to/prepare-for-production.md

```diff
@@ -0,0 +1,193 @@
+
+# Relate with Charmed PostgreSQL and Charmed Redis
+juju relate superset-k8s-beat postgresql-k8s
+juju relate superset-k8s-beat redis-k8s
+```
+
+### Scaling Applications
+Charmed Superset supports independent scaling of the web server and workers. The web server and workers can be scaled horizontally to handle more load, while the Beat scheduler should remain singular. Use the juju `scale-application` command to adjust the number of instances of each service as needed.
+
+```
+juju scale-application superset-k8s -n 3
+```
+We recommend `3` units of server and `3` units of worker applications for a production deployment.
+
+
+## Enabling Database Backups
+Superset relies on PostgreSQL for metadata storage. You can enable database backups by integrating the PostgreSQL charm with the S3 integrator. Configure it to store backups in an S3-compatible bucket, ensuring disaster recovery and data retention.
+
+[A complete how-to guide can be found here.](https://charmhub.io/postgresql-k8s/docs/h-configure-s3-aws)
+
+## Integrate with Canonical Observability Stack
+For observability, Superset can be integrated with [Canonical Observability Stack](https://charmhub.io/topics/canonical-observability-stack). This integration allows you to monitor metrics, logs, and events from Superset, enabling proactive health checks and performance analysis. Prometheus, Grafana and Loki are included as part of this observability suite.
+
+To deploy `cos-lite` and expose its endpoints as offers, follow these steps:
+
+```bash
+# Deploy the cos-lite bundle
+juju add-model cos
+juju deploy cos-lite --trust
+```
+
+```bash
+# Expose the cos integration endpoints
+juju offer prometheus:metrics-endpoint
+juju offer loki:logging
+juju offer grafana:grafana-dashboard
+
+# Relate superset to the cos-lite apps
+juju relate superset-k8s admin/cos.grafana
+juju relate superset-k8s admin/cos.loki
+juju relate superset-k8s admin/cos.prometheus
+```
+
+```bash
+# Access grafana with username "admin" and password
+juju run grafana/0 -m cos get-admin-password --wait 1m
+
+# Note the application IP address
+juju status
+```
+The Grafana UI can be found on the application IP address, port 3000. Pre-built dashboards are included under `Superset Metrics`.
+
+
+## Security
+
+### Terminate TLS at Ingress
+Superset can terminate TLS at the ingress by leveraging the [Nginx Ingress Integrator Charm](https://charmhub.io/nginx-ingress-integrator).
+
+You can deploy this with:
+
+```bash
+# Deploy the nginx ingress integrator charm
+juju deploy nginx-ingress-integrator --trust
+```
+
+#### Using K8s Secrets
+You can use a self-signed or production-grade TLS certificate stored in a Kubernetes secret. The secret is then associated with the ingress to encrypt traffic between clients and the Superset UI.
+
+For self-signed certificates you can do the below, if you have a production-grade certificate skip to the `create a k8s secret` step.
```

Comment:
```suggestion
For self-signed certificates you can do the following.  
```
And transform code comments into text with each required step

For the " if you have a production-grade certificate skip to the `create a k8s secret` step.", you can add this info as a note box below the last command.